code/mob/living/critter/flock/flockdrone.dm

```diff
@@ -1093,4 +1093,18 @@
 	var/obj/item/temp = item
 	if(temp)
 		animate(temp) // cancel animation
+		if(temp.material)
+			//code from setMaterial which sets up color + texture
+			var/set_color_alpha = TRUE
+			temp.alpha = 255
+			temp.color = null
+			temp.UpdateOverlays(null, "material")
+			if (islist(temp.mat_appearances_to_ignore) && length(temp.mat_appearances_to_ignore))
+				if (temp.material.name in temp.mat_appearances_to_ignore)
+					set_color_alpha = FALSE
+			if (set_color_alpha && temp.mat_changeappearance && temp.material.applyColor)
+				if (temp.material.texture)
+					temp.setTexture(temp.material.texture, temp.material.texture_blend, "material")
+				temp.alpha = temp.material.alpha
+				temp.color = temp.material.color
```

Comment:
Would it possible to just set the same material on the item so we don't have to copy+paste this code?

Reply:
The setmaterial code has a whole bunch of calls in it for triggering behaviour on material change. This was the only way I could see to restore appearance without hitting those triggers.

Reply:
maybe make a `set_material_appearance` proc called by both?